sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3943,6 +3943,36 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-34421: Resolve temporary functions and views in views with CTEs") {
+    withTempView("temp_view") {
+      spark.udf.register("temp_func", identity[Int](_))
+
+      sql(
+        s"""
+           |CREATE TEMPORARY VIEW temp_view AS
+           |WITH cte AS (
+           |  SELECT temp_func(0)
+           |)
+           |SELECT * FROM cte
+           |""".stripMargin)
+      checkAnswer(sql("SELECT * FROM temp_view"), Row(0))
+    }
+    withTempView("temp_view") {
+      sql("CREATE TEMPORARY VIEW temp_view AS SELECT 0")
+
+      intercept[AnalysisException] {
```

Comment:
Can we also add an assertion for the error message here?

Reply:
Sure, added in https://github.com/apache/spark/pull/31550/commits/1cd1cb34bf82248f42e6b1927c7b65c4c25944c8